network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/NetworkImpl.java

```diff
@@ -153,9 +154,21 @@ public Stream<Bus> getBusStream() {
             return getVoltageLevelStream().flatMap(vl -> vl.getBusView().getBusStream());
         }
 
+        private Map<String, Bus> getIdToBusMap() {
+            return getBusStream().collect(ImmutableMap.toImmutableMap(Bus::getId, Functions.identity()));
+        }
+
         @Override
         public Bus getBus(String id) {
-            return getBusStream().filter(b -> b.getId().equals(id)).findFirst().orElse(null);
+            // checking cached data first
+            Map<String, Bus> cachesBuses = getResource().getAttributes().getBusCache();
+            if (cachesBuses != null) {
+                return cachesBuses.get(id);
+            }
+            // if cache not existing, creating it
+            Map<String, Bus> calculatedBuses = getIdToBusMap();
```

Comment:
Rename variable by `idToBusMap`
It's not a calcul

Reply:
done